test_communication/test/requester_py.py

```diff
@@ -17,55 +17,52 @@
 import sys
 
 
-def requester(service_name, number_of_cycles):
-    from test_msgs.service_fixtures import get_test_srv
+def requester(service_name):
     import rclpy
-
-    service_pkg = 'test_msgs'
-    module = importlib.import_module(service_pkg + '.srv')
-    srv_mod = getattr(module, service_name)
-
+    from test_msgs.service_fixtures import get_test_srv
     rclpy.init(args=[])
 
-    node = rclpy.create_node('requester')
+    try:
+        # Import the service
+        service_pkg = 'test_msgs'
+        module = importlib.import_module(service_pkg + '.srv')
+        srv_mod = getattr(module, service_name)
 
-    srv_fixtures = get_test_srv(service_name)
+        srv_fixtures = get_test_srv(service_name)
+        service_name = 'test_service_' + service_name
 
-    client = node.create_client(srv_mod, 'test_service_' + service_name)
+        node = rclpy.create_node('requester')
+        try:
+            # wait for the service to be available
+            client = node.create_client(srv_mod, service_name)
+            tries = 15
+            while rclpy.ok() and not client.wait_for_service(timeout_sec=1.0) and tries > 0:
+                print('service not available, waiting again...')
+                tries -= 1
+            assert tries > 0
```

Comment:
can you please add an assert message here ? so that it's easy to now what went wrong if we have only stderr output

Reply:
Done! d5afa0e46ba37a74bf84e9b5c15fc5d6f6eca1db